cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -528,9 +546,39 @@ def gen_json_out(self):
             })
         return json.dumps(json_out)+LINESEP
 
+    def upgrade_user_permissions(self):
+        # check whether the given user exists or not
+        cmd_json = {"prefix": "auth get", "entity": self.run_as_user, "format": "json"}
+        ret_val, json_out, err_msg = self._common_cmd_json_gen(cmd_json)
+        if ret_val != 0 or len(json_out) == 0:
+            raise ExecutionFailureException("'auth get {}' command failed.\n".format(self.run_as_user) +
+                "Error: {}".format(err_msg if ret_val != 0 else self.EMPTY_OUTPUT_LIST))
+        j_first = json_out[0]
+        existing_caps = j_first['caps']
+        cap_keys = ["mon", "mgr", "osd"]
+        for eachKey in cap_keys:
+            min_cap_perm_list = [x.strip() for x in self.MIN_USER_CAP_PERMISSIONS.get(eachKey, '').split(',') if x.strip()]
+            cur_cap_perm_list = [x.strip() for x in existing_caps.get(eachKey, '').split(',') if x.strip()]
+            cur_cap_perm_list.extend(min_cap_perm_list)
```

Comment:
We need to trim dup. Do we get duplicates with `extend`?

Reply:
@leseb , yes it is possible and that is exactly what we are trying to do in the next line. We convert it to a set (and then back to list) to eliminate any duplicates.
```python
cur_cap_perm_list = list(set(cur_cap_perm_list))
```

Reply:
Ok missed that, thanks!

Reply:
Figured out that there is a small issue in using `set()` function, as it re-orders the caps' permission list.
By order what I meant was, ceph's `get-or-create` command will fail if the order of 'mon' / 'mgr' / 'osd' caps are different.

For example,
If we try to do a `get-or-create` the following existing user,
```
ceph auth get client.healthchecker2
exported keyring for client.healthchecker2
[client.healthchecker2]
        key = AQDS439fvGEgCxAAh6WBYzMU+C/Sjwi+ox7JyA==
        caps mon = "allow r, allow command quorum_status, allow command version"
```
with mon caps order changed `allow command version, allow r, allow command quorum_status`

```
ceph auth get-or-create client.healthchecker2 mon 'allow command version, allow r, allow command quorum_status'
Error EINVAL: key for client.healthchecker2 exists but cap mon does not match
```

So I have to modify the script to preserve the order (and eliminate the use of `set()` function)